src/libraries/System.Security.Cryptography.Algorithms/tests/Rfc2898OneShotTests.cs

```diff
@@ -0,0 +1,321 @@
+        {
+            foreach (HashAlgorithmName hashAlgorithm in SupportedHashAlgorithms)
+            {
+                // hashAlgorithm, length, iterations, passwordHex, saltHex
+                yield return new object[] { hashAlgorithm.Name, 1, 1, s_passwordBytes.ByteArrayToHex(), s_salt.ByteArrayToHex() };
+                yield return new object[] { hashAlgorithm.Name, 1, 1, "", s_salt.ByteArrayToHex() };
+                yield return new object[] { hashAlgorithm.Name, 257, 257, "", s_salt.ByteArrayToHex() };
+                yield return new object[] { hashAlgorithm.Name, 257, 257, "D8D8D8D8D8D8D8D8", "D8D8D8D8D8D8D8D8" };
+                yield return new object[] { hashAlgorithm.Name, 257, 257, "0000000000000000", "0000000000000000" };
+            }
+
+            // Test around HMAC SHA1 and SHA256 block boundaries
+            for (int blockBoundary = 63; blockBoundary <= 65; blockBoundary++)
+            {
+                byte[] password = new byte[blockBoundary];
+                yield return new object[] { HashAlgorithmName.SHA1.Name, 257, 257, password.ByteArrayToHex(), "0000000000000000" };
+                yield return new object[] { HashAlgorithmName.SHA256.Name, 257, 257, password.ByteArrayToHex(), "0000000000000000" };
+            }
+
+            // Test around HMAC SHA384 and SHA512 block boundaries
+            for (int blockBoundary = 127; blockBoundary <= 129; blockBoundary++)
+            {
+                byte[] password = new byte[blockBoundary];
+                yield return new object[] { HashAlgorithmName.SHA384.Name, 257, 257, password.ByteArrayToHex(), "0000000000000000" };
+                yield return new object[] { HashAlgorithmName.SHA512.Name, 257, 257, password.ByteArrayToHex(), "0000000000000000" };
+            }
+        }
+
+        public static IEnumerable<object[]> Pbkdf2_PasswordString_Compare_Data()
+        {
+            string largePassword = new string('y', 1024);
+
+            foreach (HashAlgorithmName hashAlgorithm in SupportedHashAlgorithms)
+            {
+                // hashAlgorithm, length, iterations, password, saltHex
+                yield return new object[] { hashAlgorithm.Name, 1, 1, Password, s_salt.ByteArrayToHex() };
+                yield return new object[] { hashAlgorithm.Name, 1, 1, "", s_salt.ByteArrayToHex() };
+                yield return new object[] { hashAlgorithm.Name, 257, 257, "", s_salt.ByteArrayToHex() };
+                yield return new object[] { hashAlgorithm.Name, 257, 257, Password, "D8D8D8D8D8D8D8D8" };
+                yield return new object[] { hashAlgorithm.Name, 257, 257, Password, "0000000000000000" };
+
+                // case for password exceeding the stack buffer limit.
+                yield return new object[] { hashAlgorithm.Name, 257, 257, largePassword, "0000000000000000" };
+            }
+        }
+
+        private static HashAlgorithmName[] SupportedHashAlgorithms => new []
```

Comment:
We should probably add the "official" test cases to this this test file as well. See RFC 6070, Sec. 2.

Reply:
Good idea. One of them has a high iteration count. I'll see how long it takes CI to do it (it only took my box 2 seconds to run it). If CI takes excessively long I will move that one case to its own `OuterLoop` test.

Reply:
The highest duration I saw was over 20 seconds on Windows 7, which seems iffy for inner-loop speed. I will move it to outerloop.